plugins/chat/assets/javascripts/discourse/initializers/chat-audio.js

```diff
@@ -15,6 +16,42 @@ export default {
       return;
     }
 
+    this.supportsServiceWorker = () => {
+      if (
+        !(
+          "serviceWorker" in navigator &&
+          typeof ServiceWorkerRegistration !== "undefined" &&
+          !capabilities.isAppWebview &&
+          navigator.serviceWorker.controller &&
+          navigator.serviceWorker.controller.state === "activated"
+        )
+      ) {
+        return false;
+      }
+
+      return true;
+    };
```

Comment:
This seems more complex than it should be:

```suggestion
    this.supportsServiceWorker = () => (
        "serviceWorker" in navigator &&
        typeof ServiceWorkerRegistration !== "undefined" &&
        !capabilities.isAppWebview &&
        navigator.serviceWorker.controller &&
        navigator.serviceWorker.controller.state === "activated"
      );
```